src/app/components/single-image/single-image.component.ts

```diff
@@ -54,6 +59,13 @@ export class SingleImageComponent implements OnInit {
         // modal closed
       });
   }
+
+  retrievePostDescription() {
+    this.db.database.ref(this.user.uid + '/files/' + this.imageName.split('.')[0]).once('value')
+      .then((snapshot) => {
+        this.description = snapshot.val().description;
+      });
+  }
 }
 
```

Comment:
I think we should have  a subscription variable for this.firebase.user.subscribe. Then unsubscribe for it in ngDestroy.

Reply:
done